enemies.js

```diff
@@ -238,4 +238,322 @@ class PirahnaPlant {
             }
         }
     };
-};
\ No newline at end of file
+};
+//the hammer brother enemy
+class HammerBro
+{
+    constructor(game,x,y,facing)
+    {
+        Object.assign(this, { game, x, y, facing });
+        this.velocity = { x: Math.pow(-1, this.facing)*PARAMS.BITWIDTH, y: 0 }; // pixels per second
+        this.spritesheet = ASSET_MANAGER.getAsset("./sprites/enemies.png");
+        this.animations = [];
+        this.animations.push(new Animator(this.spritesheet, 179, 89, 16, 25, 2, 0.1, 14, false, true));
+        this.animations.push(new Animator(this.spritesheet, 119, 89, 16, 24, 2, 0.1, 14, false, true));
+        this.paused = true;
+        this.dead = false;
+        this.deadCounter = 0;
+        //this.updateBB();
+        this.swap = 60;
+        this.jump = 240;
+        this.jumpType = 1;
+        this.passThrough = 0;
+        this.throw = 30;
```

Comment:
The naming convention here can be improve. When you have a variable with similar names like *throw*, *throws*, *throwing* without a clear comment, this adds noise to your code. Besides, many of your counters are frame based (the amount of update being called). As we cannot consistently get exactly 60 fps, your comments (which use "seconds") can be misleading. From what I know, an original Mario mechanism is heavily frame based, that's why we have terminology like *perfect frame jump*, etc. Thus the implementation here might not exactly be a bad thing.